internal/uvm/create.go

```diff
@@ -170,25 +250,43 @@ func Create(opts *UVMOptions) (*UtilityVM, error) {
 	}
 
 	if uvm.operatingSystem == "windows" {
-		hcsDocument.VirtualMachine.Chipset.UEFI.BootThis = &schema2.VirtualMachinesResourcesUefiBootEntryV2{
-			DevicePath: `\EFI\Microsoft\Boot\bootmgfw.efi`,
-			DiskNumber: 0,
-			UefiDevice: "VMBFS",
-		}
+		hcsDocument.VirtualMachine.Chipset.UEFI.BootThis = &schema2.VirtualMachinesResourcesUefiBootEntryV2{DevicePath: `\EFI\Microsoft\Boot\bootmgfw.efi`}
 		hcsDocument.VirtualMachine.ComputeTopology.Memory.DirectFileMappingMB = 1024 // Sensible default, but could be a tuning parameter somewhere
 		hcsDocument.VirtualMachine.Devices.VirtualSMBShares[0].Path = filepath.Join(uvmFolder, `UtilityVM\Files`)
 		hcsDocument.VirtualMachine.Devices.VirtualSMBShares[0].Flags = schema2.VsmbFlagReadOnly | schema2.VsmbFlagPseudoOplocks | schema2.VsmbFlagTakeBackupPrivilege | schema2.VsmbFlagCacheIO | schema2.VsmbFlagShareRead
 	} else {
 		hcsDocument.VirtualMachine.Devices.GuestInterface.BridgeFlags = 3 // TODO: Contants
+		hcsDocument.VirtualMachine.Devices.VirtualSMBShares[0].Path = opts.BootFilesPath
+		hcsDocument.VirtualMachine.Devices.VirtualSMBShares[0].Flags = schema2.VsmbFlagReadOnly | schema2.VsmbFlagShareRead | schema2.VsmbFlagCacheIO | schema2.VsmbFlagTakeBackupPrivilege // 0x17 (23 dec)
+		hcsDocument.VirtualMachine.Devices.VPMem = &schema2.VirtualMachinesResourcesStorageVpmemControllerV2{MaximumCount: opts.VPMemDeviceCount}
 		hcsDocument.VirtualMachine.Chipset.UEFI.BootThis = &schema2.VirtualMachinesResourcesUefiBootEntryV2{
 			DevicePath:   `\` + opts.KernelFile,
-			DiskNumber:   0,
-			UefiDevice:   "VMBFS",
-			OptionalData: `initrd=\` + opts.InitrdFile,
+			OptionalData: `initrd=\` + opts.RootFSFile,
+		}
+
+		// Support for VPMem VHD(X) booting rather than initrd..
+		if actualRootFSType == PreferredRootFSTypeVHD {
+			hcsDocument.VirtualMachine.Chipset.UEFI.BootThis.OptionalData = `root=/dev/pmem0 init=/init`
+			hcsDocument.VirtualMachine.Devices.VPMem.Devices = make(map[string]schema2.VirtualMachinesResourcesStorageVpmemDeviceV2)
+			imageFormat := "VHD1"
+			if strings.ToLower(filepath.Ext(opts.RootFSFile)) == "vhdx" {
+				imageFormat = "VHD2"
```

Comment:
The schema doesnt look like it even supports VHD2 here

Reply:
It does. It's `Default` 😕